src/broadcast/scrobblingmanager.h

```diff
@@ -0,0 +1,63 @@
+#pragma once
+
+#include <QObject>
+#include <QMutex>
+#include <QLinkedList>
+
+#include "broadcast/metadatabroadcast.h"
+#include "mixer/basetrackplayer.h"
+#include "track/track.h"
+#include "track/tracktiminginfo.h"
+
+class BaseTrackPlayer;
+class PlayerManager;
+
+class ScrobblingManager : public QObject {
+    Q_OBJECT
+  public:
+    ScrobblingManager(PlayerManager *manager);
+  private:
+    struct TrackInfo {
+        TrackPointer m_pTrack;
+        TrackTimingInfo m_trackInfo;
+        QLinkedList<QString> m_players;
+        TrackInfo(TrackPointer pTrack) : 
+        m_pTrack(pTrack), m_trackInfo(pTrack)
+        {} 
+    };
+    struct TrackToBeReset {
+        TrackPointer m_pTrack;
+        QString m_playerGroup;
+        TrackToBeReset(TrackPointer pTrack, const QString &playerGroup) :
+        m_pTrack(pTrack), m_playerGroup(playerGroup) {}
+    };   
+
+    PlayerManager *m_pManager;
+
+    MetadataBroadcaster m_broadcaster;
+
+    QMutex m_mutex;
```

Comment:
Why do you need this mutex? What exactly are the shared resources it protects and who are the contestants? I only see some slots at the interface, which is good decision.

Reply:
As far as I know, each deck can request a track to the engine buffer independently with its own thread. And each thread signals independently. I'm not sure about this since I haven't explored in depth the engine. However, if that's the case, I need to protect the list of tracks from concurrent access, don't I?

Reply:
Your slots will only be invoked by your object's event loop thread. It doesn't matter if the signal is sent from another thread.

Reply:
Hmm now that you mention it, if the decks are owned by the PlayerManager, and the manager owns the ScrobblingManager, they all live in the same thread don't they? I guess I don't need a mutex then. But I need to make absolutely sure because I don't want to find data race conditions further down the road which can be a pain in the ass to detect.

Reply:
We will finally have threads, accessing this Infos. If they will be qt threads, receiving a singnal, we can get arround the issue. 
But I think we need a qt free plain c interface for the consuming plug-ins, to get around the mess with crashing because of incompatible shared libraries...